core/survey/_surveyValidator/nodeDefValidationsValidator.js

```diff
@@ -19,11 +21,11 @@ const validate = async (survey, nodeDef, nodeDefValidations, errorKey = null) =>
   return R.pipe(
     Validation.assocFieldValidation(
       NodeDefValidations.keys.expressions,
-      await NodeDefExpressionsValidator.validate(survey, nodeDef, NodeDefValidations.getExpressions(nodeDefValidations), false)
+      await NodeDefExpressionsValidator.validate(survey, nodeDef, nodeDef, NodeDefValidations.getExpressions(nodeDefValidations), false, true)
```

Comment:
I don't understand why circular references are allowed here? (the last boolean parameter)

Reply:
because in the node def validations you can use the current node, validating for example that it's value is greater than 0 or whatever.